### ISSUE TITLE
ignore_templates isn't being respected when used with pre-commit

### ISSUE THREAD

**Issue by chizou** [aws-cloudformation/cfn-lint#1427]
```
$ cfn-lint --version
cfn-lint 0.29.0
```

When I run `cfn-lint` on the cli, it works fine and respects the `ignore_templates` in `.cfnlintrc. However, if I try to run it via pre-commit, it doesn't respect the the config file.

Here are my `.cfnlintrc` contents:
```
ignore_templates:
    - templates/template_configs.yml
    - provisioner-functions/lambdas/cf_deployer/templates/template_configs.yml
```

Here's the output from a commit:
```
$ git commit -m 'Removing cloudtrail since we have the org cloudtrail now'
[WARNING] Unstaged files detected.
[INFO] Stashing unstaged files to /home/achou/.cache/pre-commit/patch1584505754.
Detect secrets...........................................................Passed
Black................................................(no files to check)Skipped
Trim Trailing Whitespace.................................................Passed
Flake8...............................................(no files to check)Skipped
Sort Imports.............................................................Passed
AWS CloudFormation Linter................................................Failed
hookid: cfn-python-lint

E1001 Missing top level item Resources to file module
None:1:1

E1001 Top level item mfa-policy.yml isn't valid
None:11:1

E1001 Top level item governance-roles.yml isn't valid
None:23:1

[INFO] Restored changes from /home/achou/.cache/pre-commit/patch1584505754.
```

What's strange is that if I try to run it with debug output, it seems to give a different set of errors:
```
$ git commit -m 'Removing cloudtrail since we have the org cloudtrail now'
[WARNING] Unstaged files detected.
[INFO] Stashing unstaged files to /home/achou/.cache/pre-commit/patch1584505469.
Detect secrets...........................................................Passed
Black................................................(no files to check)Skipped
Trim Trailing Whitespace.................................................Passed
Flake8...............................................(no files to check)Skipped
Sort Imports.............................................................Passed
AWS CloudFormation Linter................................................Failed
hookid: cfn-python-lint

2020-03-17 22:24:31,008 - cfnlint - DEBUG - Looking for CFLINTRC before attempting to load
2020-03-17 22:24:31,008 - cfnlint - DEBUG - Found Project level CFNLINTRC
2020-03-17 22:24:31,008 - cfnlint - DEBUG - Validating User CFNLINTRC
2020-03-17 22:24:31,009 - cfnlint - DEBUG - Validating CFNLINTRC config with given JSONSchema
2020-03-17 22:24:31,009 - cfnlint - DEBUG - Schema used: {'$id': 'https://github.com/aws-cloudformation/cfn-python-lint/blob/master/src/cfnlint/data/CfnLintCli/config/schema.json', '$schema': 'http://json-schema.org/draft-07/schema#', 'description': 'CFNLINTRC configuration schema', 'title': 'CFNLINTRC JSON Schema', 'type': 'object', 'additionalProperties': False, 'properties': {'append_rules': {'description': 'Location of directories to append rules from', 'items': {'type': 'string'}, 'type': 'array'}, 'ignore_checks': {'description': 'List of checks to ignore', 'items': {'type': 'string'}, 'type': 'array'}, 'include_checks': {'description': 'List of checks to include', 'items': {'type': 'string'}, 'type': 'array'}, 'mandatory_checks': {'description': 'List of mandatory checks to enforce', 'items': {'type': 'string'}, 'type': 'array'}, 'override_spec': {'description': 'Path to spec file to override with', 'type': 'string'}, 'regions': {'description': 'Regions to test against', 'items': {'type': 'string'}, 'type': 'array'}, 'configure_rules': {'description': 'Configure rules', 'patternProperties': {'^.*$': {'type': 'object', 'patternProperties': {'^.*$': {'anyOf': [{'type': 'string'}, {'type': 'integer'}, {'type': 'boolean'}]}}}}, 'additionalProperties': False, 'type': 'object'}, 'templates': {'description': 'Templates to lint', 'items': {'type': 'string'}, 'type': 'array'}, 'ignore_templates': {'description': 'Templates to ignore', 'items': {'type': 'string'}, 'type': 'array'}}}
2020-03-17 22:24:31,009 - cfnlint - DEBUG - Config used: {}
2020-03-17 22:24:31,011 - cfnlint - DEBUG - CFNLINTRC looks valid!
2020-03-17 22:24:31,011 - cfnlint - DEBUG - Parsing CFNLINTRC
2020-03-17 22:24:31,012 - cfnlint - DEBUG - Validating Project CFNLINTRC
2020-03-17 22:24:31,012 - cfnlint - DEBUG - Validating CFNLINTRC config with given JSONSchema
2020-03-17 22:24:31,012 - cfnlint - DEBUG - Schema used: {'$id': 'https://github.com/aws-cloudformation/cfn-python-lint/blob/master/src/cfnlint/data/CfnLintCli/config/schema.json', '$schema': 'http://json-schema.org/draft-07/schema#', 'description': 'CFNLINTRC configuration schema', 'title': 'CFNLINTRC JSON Schema', 'type': 'object', 'additionalProperties': False, 'properties': {'append_rules': {'description': 'Location of directories to append rules from', 'items': {'type': 'string'}, 'type': 'array'}, 'ignore_checks': {'description': 'List of checks to ignore', 'items': {'type': 'string'}, 'type': 'array'}, 'include_checks': {'description': 'List of checks to include', 'items': {'type': 'string'}, 'type': 'array'}, 'mandatory_checks': {'description': 'List of mandatory checks to enforce', 'items': {'type': 'string'}, 'type': 'array'}, 'override_spec': {'description': 'Path to spec file to override with', 'type': 'string'}, 'regions': {'description': 'Regions to test against', 'items': {'type': 'string'}, 'type': 'array'}, 'configure_rules': {'description': 'Configure rules', 'patternProperties': {'^.*$': {'type': 'object', 'patternProperties': {'^.*$': {'anyOf': [{'type': 'string'}, {'type': 'integer'}, {'type': 'boolean'}]}}}}, 'additionalProperties': False, 'type': 'object'}, 'templates': {'description': 'Templates to lint', 'items': {'type': 'string'}, 'type': 'array'}, 'ignore_templates': {'description': 'Templates to ignore', 'items': {'type': 'string'}, 'type': 'array'}}}
2020-03-17 22:24:31,012 - cfnlint - DEBUG - Config used: {'ignore_templates': ['provisioner-functions/lambdas/cf_deployer/templates/template_configs.yml']}
2020-03-17 22:24:31,014 - cfnlint - DEBUG - CFNLINTRC looks valid!
2020-03-17 22:24:31,014 - cfnlint - DEBUG - User configuration loaded as
2020-03-17 22:24:31,015 - cfnlint - DEBUG - {}
2020-03-17 22:24:31,015 - cfnlint - DEBUG - Project configuration loaded as
2020-03-17 22:24:31,015 - cfnlint - DEBUG - {'ignore_templates': ['provisioner-functions/lambdas/cf_deployer/templates/template_configs.yml']}
2020-03-17 22:24:31,015 - cfnlint - DEBUG - Merging configurations...
2020-03-17 22:24:31,015 - cfnlint - DEBUG - Begin linting of file: None
2020-03-17 22:24:31,015 - cfnlint.decode - ERROR - Template file not found: None
2020-03-17 22:24:31,016 - cfnlint - DEBUG - Completed linting of file: None
E0000 Template file not found: None
None:1:1

[INFO] Restored changes from /home/achou/.cache/pre-commit/patch1584505469.
```

**Comment by PatMyron** [comment]
@kddejong duplicate of https://github.com/aws-cloudformation/cfn-python-lint/issues/987?

**Comment by chizou** [comment]
> @kddejong duplicate of #987?

Interesting. My output in the second example with debug output seems to mirror #987. But in the first one, it's actually linting the file, ie: not respecting `ignore_templates`. I'd say that's a different bug. Not to mention producing two different outputs (debug vs. non-debug) might be another bug?

**Comment by kddejong** [comment]
This also acts differently.  If I add args it it changes how it works.
```
repos:
-   repo: https://github.com/aws-cloudformation/cfn-python-lint
    rev: v0.42.0   # The version of cfn-lint to use
    hooks:
    -   id: cfn-python-lint
        files: .*\.(json|yml|yaml)$
        args:
        - --debug
```

**Comment by kddejong** [comment]
This seems to be working the way I would hope it would.
```
repos:
-   repo: https://github.com/aws-cloudformation/cfn-python-lint
    rev: v0.42.0   # The version of cfn-lint to use
    hooks:
    -   id: cfn-python-lint
```

**Comment by kddejong** [comment]
My current recommendation is to use .cfnlintrc or the pre-commit configuration exclusively. Mixing and matching has the possibility of causing impacts. So I would recommend either switching `.pre-commit-hooks.yaml` to:
```yaml
repos:
  - repo: https://github.com/aws-cloudformation/cfn-lint/
    rev: v0.69.1 # The version of cfn-lint to use
    hooks:
      - id: cfn-python-lint
        name: AWS CloudFormation Linter
        exclude: ^(templates/template_configs.yml|provisioner-functions/lambdas/cf_deployer/templates/template_configs.yml)$
```
or use a `.cfnlintrc` file of
```yaml
ignore_templates:
  - templates/template-config.yaml
  - provisioner-functions/lambdas/cf_deployer/templates/template_configs.yml
templates:
  - '**/*.yaml'

```
and a `.pre-commit-hooks.yaml` of
```yaml
repos:
  - repo: https://github.com/aws-cloudformation/cfn-lint/
    rev: v0.69.1 # The version of cfn-lint to use
    hooks:
      - id: cfn-python-lint
        name: AWS CloudFormation Linter
        pass_filenames: false
```

**Comment by kddejong** [comment]
I haven't been able to reproduce the 1st output provided above.

**Comment by kddejong** [comment]
If we can get more details to reproduce the 1st output I'll re-open this PR.  We have updated our guidance of usage with `.cfnlintrc` files.
